### ISSUE TITLE
Pyrotech workbench item placement out of sync

### ISSUE THREAD

**Issue by krageon** [jrbudda/Vivecraft_112#146]
When targeting a pyrotech workbench (it's weird on other "crafting with placement" blocks also, but the workbench is the most obvious), the workbench will light up with an image of the item you are trying to place in the square you are aiming at. In almost every case, the item will not be placed or it will be placed in the wrong slot when right clicking. It appears that there is some sort of complex interplay between the left (holding the GUI) and right (holding the item) hands, because I was able to mess around with moving my left hand around and eventually got it to place an item in the right slot. Removing items is similarly difficult.

If I had to guess, something is going wrong in the raytracing code (I'm assuming that this is what's used to determine what part of the block is being targeted): https://github.com/codetaylor/pyrotech/blob/5e6c45402e7143ff649571c6a54086fb4795b821/src/main/java/com/codetaylor/mc/pyrotech/modules/tech/basic/block/spi/BlockWorktableBase.java#L49

Although I can't really figure out why the right square lights up with the right item, but the wrong square is still selected when resolving the interaction on right click.

**Comment by Techjar** [comment]
Yeah that's weird, we set the player's gaze to look at where the crosshair is, so it should be working. The left hand also shouldn't have any effect on it.

**Comment by talchas** [comment]
Try v10, it looks like the most likely cause should be fixed now.
